tracing-opentelemetry/src/main/java/io/micronaut/tracing/opentelemetry/DefaultOpenTelemetryFactory.java

```diff
@@ -98,4 +101,11 @@ protected OpenTelemetry defaultOpenTelemetry(ApplicationConfiguration applicatio
 
     }
 
+    @PreDestroy
+    void resetForTest(Environment environment) {
```

Comment:
I don't think we should add a method just for tests. 

Maybe it works to create a new class in the test class path. 

`@Replaces(DefaultOpenTelemtryFactory.clas)`
class DefaultOpenTelemtryFactoryReplacement extends DefaultOpenTelemtryFactory. { 
 @PreDestroy
     void resetForTest(Environment environment) {
....
}
}
````

Reply:
@sdelamo  I don't know what is the best approach here. Issue is in the user application tests not in the module tests. I can create a new class `DefaultOpenTelemtryFactoryReplacement` that can have `@Replaces(DefaultOpenTelemtryFactory.class)` and `@Requires(env = Environment.TEST)` ? It has to be in user application classpath

Reply:
Let's leave it as you created it.